File: ios/Classes/FlutterW2aPlugin.m
#import "FlutterW2aPlugin.h"
#import "hm.h"

@implementation FlutterW2aPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"flutter_w2a"
            binaryMessenger:[registrar messenger]];
  FlutterW2aPlugin* instance = [[FlutterW2aPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else if ([@"useFingerPrinting" isEqualToString:call.method]) {
    BOOL isEnabled = [call.arguments[@"isEnabled"] boolValue];
    [hm useFingerPrinting: isEnabled];
    result(@"");
  } else if ([@"init" isEqualToString:call.method]) {
    NSString *gateway = call.arguments[@"gateway"];
    NSString *installEventName = call.arguments[@"installEventName"];
    NSLog(@"调用了初始化");
    [hm init: gateway InstallEventName: installEventName success:^(NSArray * _Nonnull array) {
      NSLog(@"初始化完毕");
      result(array);
    }];
  } else if ([@"purchase" isEqualToString:call.method]) {
    NSString *name = call.arguments[@"name"];
    NSString *currency = call.arguments[@"currency"];
    NSString *value = call.arguments[@"value"];
    NSString *contentType = call.arguments[@"contentType"];
    NSString *contentIds = call.arguments[@"contentIds"];
    [hm Purchase:name Currency:currency Value:value ContentType:contentType ContentIds:contentIds];
    result(@"");
  } else if ([@"eventPost" isEqualToString:call.method]) {
    NSString *name = call.arguments[@"name"];
    NSString *currency = call.arguments[@"currency"];
    NSString *value = call.arguments[@"value"];
    NSString *contentType = call.arguments[@"contentType"];
    NSString *contentIds = call.arguments[@"contentIds"];
    [hm EventPost:name Currency:currency Value:value ContentType:contentType ContentIds:contentIds];
    result(@"");
  } else if ([@"userDataUpdateEvent" isEqualToString:call.method]) {
    NSString *email = call.arguments[@"email"];
    NSString *fbLoginId = call.arguments[@"fbLoginId"];
    NSString *userId = call.arguments[@"userId"];
    NSString *phone = call.arguments[@"phone"];
    [hm UserDataUpdateEvent:email Fb_login_id:fbLoginId UserId:userId Phone:phone];
    result(@"");
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: android/src/main/java/com/example/flutter_w2a/FlutterW2aPlugin.java
package com.example.flutter_w2a;

import androidx.annotation.NonNull;

import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import com.huntmobi.web2app.hm;
import com.huntmobi.web2app.utils.DataCallback;
import com.huntmobi.web2app.utils.NetCallback;
import com.huntmobi.web2app.utils.NetInfo;
import com.huntmobi.web2app.utils.OutNetCallback;
import android.app.Application;
import android.content.Context;
import java.util.Arrays;
import android.util.Log;

/** FlutterW2aPlugin */
public class FlutterW2aPlugin implements FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private MethodChannel channel;
  private Context mContext;

  @Override
  public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
    channel = new MethodChannel(flutterPluginBinding.getBinaryMessenger(), "flutter_w2a");
    channel.setMethodCallHandler(this);
    mContext = flutterPluginBinding.getApplicationContext();
  }

  @Override
  public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
    Log.e("W2A", "草泥马==" + call.method);
    if (call.method.equals("getPlatformVersion")) {
      result.success("Android " + android.os.Build.VERSION.RELEASE);
    } else if (call.method.equals("useFingerPrinting")) {
      boolean isEnabled = call.argument("isEnabled");
      hm.useFingerPrinting(isEnabled);
      result.success("");
    } else if (call.method.equals("init")) {
      String gateway = call.argument("gateWay");
      String installEventName = call.argument("installEventName");
      Application application = (Application) mContext.getApplicationContext();
      hm.Init(application, gateway, installEventName, new DataCallback() {
          @Override
          public void CallbackDealwith(String[] strings) {
              if (strings != null){
                result.success(Arrays.asList(strings));
              }
          }
      });
    } else if (call.method.equals("purchase")) {
      String name = call.argument("name");
      String currency = call.argument("currency");
      String value = call.argument("value");
      String contentType = call.argument("contentType");
      String contentIds = call.argument("contentIds");
      hm.Purchase(name, currency, value, contentType, Arrays.asList(contentIds.split(",")), new OutNetCallback() {
          @Override
          public void callbackDealwith(Object info) {

          }
      });
      result.success("");
    } else if (call.method.equals("eventPost")) {
      String name = call.argument("name");
      String currency = call.argument("currency");
      String value = call.argument("value");
      String contentType = call.argument("contentType");
      String contentIds = call.argument("contentIds");
      hm.EventPost("", name, currency, value, contentType, Arrays.asList(contentIds.split(",")), new OutNetCallback() {
          @Override
          public void callbackDealwith(Object info) {

          }
      });
      result.success("");
    } else if (call.method.equals("userDataUpdateEvent")) {
      String email = call.argument("email");
      String fbLoginId = call.argument("fbLoginId");
      String userId = call.argument("userId");
      String phone = call.argument("phone");
      hm.UserDataUpdate(email, fbLoginId, phone, new OutNetCallback() {
          @Override
          public void callbackDealwith(Object info) {

          }
      });
      result.success("");
    } else {
      result.notImplemented();
    }
  }

  @Override
  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
    channel.setMethodCallHandler(null);
  }
}
